$NetBSD: patch-Makefile,v 1.5 2024/10/16 04:25:32 thorpej Exp $

Honor CFLAGS/LDFLAGS.

Allow the default location of the X16 ROM to be overridden.

Allow GIT_REV to be overridden.

--- Makefile.orig	2024-10-16 03:56:27.312106336 +0000
+++ Makefile	2024-10-16 03:57:42.190887631 +0000
@@ -16,9 +16,9 @@ else ifeq ($(OS),Windows_NT)
 	TARGET_WIN32=yes
 endif
 
-CFLAGS=-std=c99 -O3 -Wall -Werror -g $(shell $(SDL2CONFIG) --cflags) -Isrc/extern/include
-CXXFLAGS=-std=c++17 -O3 -Wall -Werror -Isrc/extern/ymfm/src
-LDFLAGS=$(shell $(SDL2CONFIG) --libs) -lm -lz
+CFLAGS+=-std=c99 -O3 -Wall -Werror -g $(shell $(SDL2CONFIG) --cflags) -Isrc/extern/include
+CXXFLAGS+=-std=c++17 -O3 -Wall -Werror -Isrc/extern/ymfm/src
+LDFLAGS+=$(shell $(SDL2CONFIG) --libs) -lm -lz
 
 # build with link time optimization
 ifndef NOLTO
@@ -36,10 +36,16 @@ ifdef TRACE
 	CFLAGS+=-D TRACE
 endif
 
+ifdef ROM_RELATIVE_PATH
+	CFLAGS+=-DROM_RELATIVE_PATH=\"$(ROM_RELATIVE_PATH)\"
+endif
+
 X16_OUTPUT=x16emu
 MAKECART_OUTPUT=makecart
 
+ifndef GIT_REV
 GIT_REV=$(shell git diff --quiet && /bin/echo -n $$(git rev-parse --short=8 HEAD || /bin/echo "00000000") || /bin/echo -n $$( /bin/echo -n $$(git rev-parse --short=7 HEAD || /bin/echo "0000000"); /bin/echo -n '+'))
+endif
 
 CFLAGS+=-D GIT_REV='"$(GIT_REV)"'
 
